Updated views, team 2

Irrialite opened the pull request:
New pull request for class based views.

Irrialite committed:
Converted views to class based generic views.

Irrialite committed:
Cleaned up, added context processor for search engine. Saves hassle.

Irrialite committed:
Cleaned up one more thing.

Irrialite committed:
Bugfix: redirects if user is already registered upon opening registration form.

Irrialite committed:
Added global context initialization, hardcoded google engine into the templates.

Irrialite committed:
Done everything except the issues I had replied to.

Irrialite committed:
Some cleanup. Called super() wherever possible.

Irrialite committed:
Last clean-up. Probably.

Irrialite committed:
Merge branch 'issue-7' into HEAD

Conflicts:
	piplmesh/frontend/templates/header.html
	piplmesh/frontend/views.py
	piplmesh/urls.py

Irrialite committed:
Merge branch 'issue-7-merged' into issue-7

Conflicts:
	piplmesh/context_processors.py
	piplmesh/frontend/templates/header.html
	piplmesh/frontend/views.py

Merge with main repository, so I can pull.

Irrialite committed:
Tiny edit. Forgot to mention a bugfix from previous commit. Page would crash if they clicked cancel on facebook app integration page.

Irrialite committed:
RegisterView:
get redirect -> dispatch

FacebookLoginView:
get redirect -> get_redirect_url

Irrialite committed:
TODO:
django 1.4 (don't know how to install it with pip)
csrf
post

Commit:
Added django development version (1.4) instead of stabile one (1.3)

Irrialite committed:
Minor edits. No clue how to do those posts instead of gets.

Irrialite committed:
Merge branch 'issue-7' of https://github.com/mmiha/PiplMesh into issue-7

Irrialite committed:
Facebook button is now a POST form. Messed with login button as well so it looks like a link.

Irrialite commented:
Django 1.4 works it seems. Didn't work before.

Irrialite committed:
Updated views to use reverse_lazy(), new in django 1.4. Django 1.4 required.

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -6,10 +6,10 @@
     {% csrf_token %}
     {{ form.as_p }}
-    <input type="submit" value="{% trans "Login" %}" />
+    <p><button type="submit" class="login_button" value="value"><span class="login_button_text">{% trans "Login" %}</span></button></p>
```
Why you need span around text? Why not simply style button element? Why additional element?

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,75 @@
+            'scope': settings.FACEBOOK_SCOPE,
+            'redirect_uri': self.request.build_absolute_uri(urlresolvers.reverse('facebook_callback')),
+        }
+        url = 'https://www.facebook.com/dialog/oauth' % kwargs
```
I am not sure what are you doing here with `kwargs`?

Irrialite replied:
Nothing, I kept it because the original method had it and I thought it would be a nice reminder that it is there in the original method even if it does nothing.

mitar replied:
I really like your argument and completely agree that we should just restructure the views and not change the code itself. But not too much. In this case it is something plain wrong. ;-) Or at least not explained. So I would fix it.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,75 @@
+            'redirect_uri': self.request.build_absolute_uri(urlresolvers.reverse('facebook_callback')),
+        }
+        url = 'https://www.facebook.com/dialog/oauth' % kwargs
+        return "%s?%s" % (url, urllib.urlencode(args))
```
Can you do keyword-based string interpolation?

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,75 @@
+        
+class FacebookLogoutView(generic_views.RedirectView):
     """ 
     Log user out of Facebook and redirect to FACEBOOK_LOGOUT_REDIRECT. 
```
_This view logs the user ..._, be consistent ...

Irrialite replied:
This was copy paste from the previous version not done by me; will change.

mitar replied:
But you have changed the comment for the other view. I am saying just that if you change one, change also the other. Be consistent.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,75 @@
+    permanent = False
+    url = settings.FACEBOOK_LOGOUT_REDIRECT
+
+    def get(self, request, *args, **kwargs):
```
It should be POST.

mitar replied:
See how your friend is doing this in #77.

Irrialite replied:
This class is actually never used. When a user logs out, they go to "logout" url not "facebook_logout".

mitar replied:
Very good observation. Later on we will combine this into one URL. But for now change the template so that Facebook logout URL is displayed if the user is logged in through Facebook, no?

Irrialite replied:
I don't know if that makes much sense since they're essentially logged out only from our site - not facebook. A message like "You have been logged out of facebook" would make no sense, since they wouldn't be logged out of facebook (they can still go on facebook and do w/e) but only from our site that is linked to facebook. I don't think the distinction is necessary.

mitar replied:
I am not saying that the user should see any difference? I am just saying that the logout URL to which user goes once it click the "Logout" link should be different based on the way the user has logged into our site, no? Why we then have the facebook logout view in the first place? It seem to me there is no need for that anyway? It does not do anything more than a normal logout view, no? So maybe we could simply remove this view and always use just the normal/standard logout view?

Irrialite replied:
Yep, keeping it is a waste of space at the moment. If we wanted to do something else depending on whether user was logged in through facebook then we could always add it back in.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,75 @@
+    permanent = False
+    url = settings.FACEBOOK_LOGIN_REDIRECT
+
+    def get(self, request, *args, **kwargs):
```
It should be POST.

mitar replied:
Are you sure Facebook is doing GET callback? Or it can also do the POST callback?

mitar replied:
OK, Facebook makes a redirect, so it is a GET.

Irrialite replied:
That's why I was confused since there's no way to get post from an outside source if they don't post.

mitar replied:
OK. But elsewhere use POST.

You should probably also add some security. So that if somebody tricks user in opening this URL without him pressing login button before, nothing happens. And because login button will be secured with POST ...

mitar replied:
So leave it for now like this. Just add a TODO comment that we should check if the callback is really called as a consequence of the user pressed Facebook login button before.

mitar commented on piplmesh/frontend/static/piplmesh/css/style.css:
```diff
@@ -65,6 +65,21 @@ ul.errorlist {
+
+/* -------------------- LOGIN ------------------------ */
+.login_button {
+	background: none;
```
Indentation!

mitar commented on piplmesh/frontend/templates/header.html:
```diff
@@ -31,7 +31,10 @@
                 <p><a href="{% url login %}">{% trans "Log in" %}</a></p>
-                <p><a class="fb_login_button" href="{% url facebook_login %}"><span class="fb_button_text">{% trans "Log in with Facebook" %}</span></a></p>
+                <form method="post" action="{% url facebook_login %}" id="fb_login_form">
+                	{% csrf_token %}
```
Indentation!

mitar commented on piplmesh/frontend/templates/header.html:
```diff
@@ -31,7 +31,10 @@
                 <p><a href="{% url registration %}">{% trans "Register" %}</a></p>
                 <p><a href="{% url login %}">{% trans "Log in" %}</a></p>
-                <p><a class="fb_login_button" href="{% url facebook_login %}"><span class="fb_button_text">{% trans "Log in with Facebook" %}</span></a></p>
+                <form method="post" action="{% url facebook_login %}" id="fb_login_form">
```
So you do know how to do this. So do it also for logout.

Irrialite replied:
Will do

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,75 @@
+            # message user that they have been logged in (maybe this will be already in auth.login once we move to MongoDB
+            return super(FacebookCallbackView, self).get(request, *args, **kwargs)
+        else:
+            # message user that they have not been logged in because they cancelled the Facebook
```
Add `TODO` at the beginning of the comment:

```
# TODO: Message user that they have not been logged in because they cancelled the Facebook
```

Also, use information provided from the Facebook why login was not successful.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,75 @@
+        if 'code' in request.GET:
+            user = auth.authenticate(token=request.GET['code'], request=request)
+            auth.login(request, user)
+            # message user that they have been logged in (maybe this will be already in auth.login once we move to MongoDB
```
Add TODO.

Irrialite committed:
Added another form for log-out, changed facebook_logout redirect to post, added keyword-based interpolation, added indentations, and added TODOs.

Didn't change the css styling for facebook button. Someone with more css knowledge should do that.

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -6,10 +6,10 @@
     {% csrf_token %}
     {{ form.as_p }}
-    <input type="submit" value="{% trans "Login" %}" />
+    <p><button type="submit" class="auth_button" value="value"><span class="auth_button_text">{% trans "Login" %}</span></button></p>
```
Span is unnecessary.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,76 @@
+            'scope': settings.FACEBOOK_SCOPE,
+            'redirect_uri': self.request.build_absolute_uri(urlresolvers.reverse('facebook_callback')),
+        }
+        url = 'https://www.facebook.com/dialog/oauth'
```
Why just do not move this to the string bellow?

`"https://www.facebook.com/dialog/oauth?%(args)s"`

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,76 @@
+            'redirect_uri': self.request.build_absolute_uri(urlresolvers.reverse('facebook_callback')),
+        }
+        url = 'https://www.facebook.com/dialog/oauth'
+        return "%(url)s?%(args)s" % {'url':url, 'args':urllib.urlencode(args)}
```
Space after `:`.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,76 @@
+        url = 'https://www.facebook.com/dialog/oauth'
+        return "%(url)s?%(args)s" % {'url':url, 'args':urllib.urlencode(args)}
+        
+class FacebookLogoutView(generic_views.RedirectView):
```
So this view is useless. We can just remove it. Also remove the setting.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,76 @@
+        if 'code' in request.GET:
+            user = auth.authenticate(token=request.GET['code'], request=request)
+            auth.login(request, user)
+            # TODO: message user that they have been logged in (maybe this will be already in auth.login once we move to MongoDB
```
Upper case the first letter of the TODO comment.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,76 @@
+            # TODO: message user that they have been logged in (maybe this will be already in auth.login once we move to MongoDB
+            return super(FacebookCallbackView, self).get(request, *args, **kwargs)
+        else:
+            # TODO: message user that they have not been logged in because they cancelled the Facebook
```
Upper case the first letter of the TODO comment.

mitar commented on piplmesh/account/views.py:
```diff
@@ -3,58 +3,76 @@
+            return super(FacebookCallbackView, self).get(request, *args, **kwargs)
+        else:
+            # TODO: message user that they have not been logged in because they cancelled the Facebook
+            # TODO: use information provided from facebook as to why login was not successful
```
Upper case the first letter of the TODO comment.

mitar commented:
This is starting to look good. Please update to the latest from the official repository so that it can be automatically merged.

Irrialite committed:
Removed facebook logout view, a few corrections.

Irrialite committed:
Merge remote-tracking branch 'remotes/top_master/master' into issue-7

Conflicts:
	piplmesh/account/templates/login.html
	piplmesh/frontend/static/piplmesh/css/style.css
	piplmesh/frontend/templates/header.html

Converted login to post where it wasn't.

Irrialite committed:
Tiny edit.

Irrialite committed:
Consistencies.

mitar commented on requirements.txt:
```diff
@@ -1 +1,2 @@
-Django==1.3.1
+Django==1.4
+
```
Empty line?

mitar commented on piplmesh/frontend/templates/header.html:
```diff
@@ -28,10 +28,17 @@
                 <p><a href="{% url login %}">{% trans "Log in" %}</a></p>
+                <form method="post" action="{% url facebook_login %}" id="fb_login_form">
+                    {% csrf_token %}
+	                  <p><button type="submit" class="fb_login_button" value="value"><span class="fb_button_text">{% trans "Log in with Facebook" %}</span></button></p>
```
Why span here?

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -11,12 +11,17 @@
+                <li>
+                		<form method="post" action="{% url facebook_login %}" id="fb_login_form2">
+                    		{% csrf_token %}
+	                  		<button type="submit" class="fb_login_button" value="value"><span class="fb_button_text">{% trans "Log in with Facebook" %}</span></button>
```
Why span?

Is this XHTML valid?

Irrialite replied:
What else could be there, other than span?

mitar replied:
Simply nothing? Why there should be span, when you already have `button` around? Just put all wanted style on `button`. No?

Irrialite replied:
I think it needs 2 divs but I wasn't the one who made it so I'm not sure.

mitar commented on piplmesh/frontend/templates/header.html:
```diff
@@ -28,10 +28,17 @@
             {% else %}
                 <p><a href="{% url registration %}">{% trans "Register" %}</a></p>
                 <p><a href="{% url login %}">{% trans "Log in" %}</a></p>
+                <form method="post" action="{% url facebook_login %}" id="fb_login_form">
```
Didn't we said that there will be no login with Facebook button on all pages but just on the login subpage?

mitar commented:
Please upgrade to the latest from main repository so that automatic merge can be done.

Irrialite committed:
XHTML valid, removed login button.

Irrialite commented:
It's upgraded to the latest.

Irrialite committed:
More specific <p> css.

mitar commented on piplmesh/account/templates/login.html:
```diff
@@ -11,12 +11,17 @@
+                <li>
+                		<form method="post" action="{% url facebook_login %}" id="fb_login_form2">
+                    		{% csrf_token %}
+	                  		<p><button type="submit" class="fb_login_button" value="value"><span class="fb_button_text">{% trans "Log in with Facebook" %}</span></button></p>
```
Unnecessary span.

mitar commented:
It is still not updated to the latest (as I am being told by GitHub)?

Irrialite commented:
I merged with the main repository and removed conflicts, thought that was enough?

mitar commented:
But this was done last time 19 hours ago, no? There is newer stuff in the main repository. See the list of commits for this pull request.

Irrialite commented:
Oh, I might have forgotten to refresh.

Irrialite committed:
Merge remote-tracking branch 'remotes/top_master/master' into issue-7

Conflicts:
	piplmesh/frontend/static/piplmesh/css/style.css
	piplmesh/frontend/views.py
	piplmesh/urls.py
